src/telegram_bot.rs

```diff
@@ -0,0 +1,91 @@
+use crate::toxic_bot::ToxicBot;
+use std::error::Error;
+use telegram_bot::*;
+use futures::StreamExt;
+use tokio;
+use regex::Regex;
+
+pub struct TelegramBot<'a> {
+    bot: &'a mut ToxicBot,
+    api: Api,
+    bot_id: UserId,
+    url_regex: Regex,
+}
+
+impl TelegramBot<'_> {
+    pub fn new(bot: &mut ToxicBot, token: String) -> Result<TelegramBot, Box<dyn Error>> {
+        let api = Api::new(token);
+        let url_regex = Regex::new(r"(http://www\.|https://www\.|http://|https://)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?")?;
+
+        Ok(TelegramBot {
+            bot,
+            api,
+            bot_id: UserId::new(0),
+            url_regex,
+        })
+    }
+
+    #[tokio::main]
+    pub async fn run(&mut self) -> Result<(), Box<dyn Error>> {
+        self.bot_id = self.api.send(GetMe).await?.id;
```

Comment:
What about to find `bot_id` in `new`?

Reply:
It was too complicated to make `new` an async function wrapped with tokio, it couldn't figure out lifetime of `bot: &mut ToxicBot`. I agree it would be better to initialize everything there, but I don't really want to spend time on it.